state/factory/factory.go

```diff
@@ -545,11 +545,13 @@ func (sf *factory) States(opts ...protocol.StateOption) (uint64, state.Iterator,
 	if cfg.Key != nil {
 		return sf.currentChainHeight, nil, errors.Wrap(ErrNotSupported, "Read states with key option has not been implemented yet")
 	}
-	values, err := readStates(sf.dao, cfg.Namespace, cfg.Keys)
+	keys, values, err := readStatesAndKeys(sf.dao, cfg.Namespace, cfg.Keys)
```

Comment:
`StatesAndKeys` related changes can become a separate PR to make it clear, and expedite the PR process